.pfnci/run.sh

```diff
@@ -0,0 +1,85 @@
+#!/bin/bash
+# run.sh is a script to run unit tests inside Docker.  This should be injected
+# by and called from script.sh.
+#
+# Usage: run.sh [target]
+# - target is a test target (e.g., "py37").
+#
+# Environment variables:
+# - GPU (default: 0) ... Set a number of GPUs to GPU.
+#       CAVEAT: Setting GPU>0 disables non-GPU tests, and setting GPU=0 disables
+#               GPU tests.
+
+set -eux
+
+cp -a /src /chainerrl
+mkdir -p /chainerrl/.git
+cd /
+
+# Remove pyc files.  When the CI is triggered with a user's local files, pyc
+# files generated on the user's local machine and they often cause failures.
+find /chainerrl -name "*.pyc" -exec rm -f {} \;
+
+TARGET="$1"
+: "${GPU:=0}"
+: "${XPYTEST_NUM_THREADS:=$(nproc)}"
+: "${PYTHON=python3}"
+: "${CHAINER=}"
+
+# Use multi-process service to prevent GPU flakiness caused by running many
+# processes on a GPU.  Specifically, it seems running more than 16 processes
+# sometimes causes "cudaErrorLaunchFailure: unspecified launch failure".
+if (( GPU > 0 )); then
+    nvidia-smi -c EXCLUSIVE_PROCESS
+    nvidia-cuda-mps-control -d
+fi
+
+################################################################################
+# Main function
+################################################################################
+
+main() {
+  marker='not slow'
+  if (( !GPU )); then
+    marker+=' and not gpu'
+    bucket=1
+  else
+    marker+=' and gpu'
+    bucket="${GPU}"
+  fi
+
+  xpytest_args=(
+      --python="${PYTHON}" -m "${marker}"
+      --bucket="${bucket}" --thread="$(( XPYTEST_NUM_THREADS / bucket ))"
+      --hint="/chainerrl/.pfnci/hint.pbtxt"
+  )
+
+  apt-get update -q
+  apt-get install -qy --no-install-recommends \
+      "${PYTHON}-dev" "${PYTHON}-pip" "${PYTHON}-setuptools" \
+      zlib1g-dev make cmake g++ git
+
+  if [ "${CHAINER}" != '' ]; then
+    "${PYTHON}" -m pip install "chainer==${CHAINER}"
+  fi
+
+  if [ "${PYTHON}" == 'python' ]; then
+    "${PYTHON}" -m pip install \
+        'cython==0.28.0' 'numpy<1.10' 'scipy<0.19' 'more-itertools<=5.0.0'
+  fi
+
+  # TODO(chainerrl): Enable ChainerRL to work without -e.
```

Comment:
Does current ChainerRL have any problem without `-e`?

Reply:
I don't know what happens exactly, but https://ci.preferred.jp/chainerrl.py3.cpu/17285/ is the result of running without `-e`. Probably, some files are not exported correctly.

Reply:
The error message `TypeError: train_agent_async() got an unexpected keyword argument 'eval_n_steps'` seems to imply that it uses an old version of chainerrl.

Reply:
I successfully run the test without `-e`. The only change from d9b6c21c09fa132073f8f8d664649009d4ae6831 is removing `-e`, so I'm not sure why it failed when you tried. https://github.com/chainer/chainerrl/pull/478

Reply:
```
diff --git a/.pfnci/run.sh b/.pfnci/run.sh
index c3d16eb..c7a7b77 100644
--- a/.pfnci/run.sh
+++ b/.pfnci/run.sh
@@ -68,8 +68,7 @@ main() {
         'cython==0.28.0' 'numpy<1.10' 'scipy<0.19' 'more-itertools<=5.0.0'
   fi

-  # TODO(chainerrl): Enable ChainerRL to work without -e.
-  "${PYTHON}" -m pip install -e /chainerrl
+  "${PYTHON}" -m pip install  /chainerrl
   # TODO(chainerrl): Prepare test target instead.
   "${PYTHON}" -m pip install \
       'pytest==4.1.1' 'pytest-xdist==1.26.1' 'mock' \
```
This is the diff. Can you check it again?

Reply:
I'm sorry, but I found my repository has cache of egg files... It is resolved.